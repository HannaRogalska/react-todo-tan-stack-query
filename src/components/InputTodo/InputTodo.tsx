import { useState, type ChangeEvent } from "react"
import {addTodo} from "../../services/todoApi"

const InputTodo = () => {
    const [inputValue, setInputValue] = useState('')
    console.log(inputValue);
    const addNewTodo = addTodo()
  return (
    <div>
      <input
        type="text"
        value={inputValue}
        onChange={(e: ChangeEvent<HTMLInputElement>) =>
          setInputValue(e.target.value)
        }
      ></input>
          <button onClick={() => addNewTodo.mutate( inputValue)}>Add todo</button>
    </div>
  );
}

export default InputTodo
