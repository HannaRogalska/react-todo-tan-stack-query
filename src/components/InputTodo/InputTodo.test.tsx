import { describe, it, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import InputTodo from "./InputTodo";
import userEvent from "@testing-library/user-event"
import { QueryClientProvider, QueryClient } from "@tanstack/react-query";

describe("Input-Todo", () => {
    it("renders input and button", () => {
        const queryClient = new QueryClient();
        render(
            <QueryClientProvider client={queryClient}>
                <InputTodo />
            </QueryClientProvider>
        );
        expect(screen.getByRole("textbox")).toBeInTheDocument();
        expect(screen.getByRole("button")).toBeInTheDocument();
    });
    it("allows typing and submitting", async () => {
        const queryClient = new QueryClient();
        render(
          <QueryClientProvider client={queryClient}>
            <InputTodo />
          </QueryClientProvider>
        );
        const input = screen.getByRole("textbox");
        const button = screen.getByRole("button");

        await userEvent.type(input, "test");
        
        expect(input).toHaveValue("test")
        await userEvent.click(button);
         expect(input).toHaveValue("");
       screen.debug()
    })
    it("snapShot", () => {
        const queryClient = new QueryClient();
        const { container } = render(
            
          <QueryClientProvider client={queryClient}>
            <InputTodo />
          </QueryClientProvider>
        );
        expect(container).toMatchSnapshot()
    })
});
