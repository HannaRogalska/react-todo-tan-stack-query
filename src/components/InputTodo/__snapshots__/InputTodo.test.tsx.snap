// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input-Todo > snapShot 1`] = `
<div>
  <form>
    <input
      class="text-center w-[350px] border border-solid rounded-lg outline-none mr-2 "
      name="title"
    />
    <input
      class="text-center w-[100px] border border-solid rounded-lg outline-none"
      type="submit"
      value="Add Todo"
    />
  </form>
</div>
`;
